packages/@glimmer/reference/lib/reference.ts

```diff
@@ -59,22 +59,22 @@ export class ReferenceCache<T> implements Tagged {
   }
 
   peek(): T {
-    if (!this.initialized) {
+    if (this.initialized === false) {
       return this.initialize();
     }
 
     return this.lastValue as T;
   }
 
   revalidate(): Validation<T> {
-    if (!this.initialized) {
+    if (this.initialized === false) {
       return this.initialize();
     }
 
     let { reference, lastRevision } = this;
     let tag = reference.tag;
 
-    if (validate(tag, lastRevision as number)) return NOT_MODIFIED;
+    if (validate(tag, lastRevision as number) === true) return NOT_MODIFIED;
```

Comment:
does validate not always return a boolean? if validate always returns a boolean then there is no benefit from this